File: stage1/problems/problem20/starter.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream> // For stringstream if needed for parsing

void solve() {
    // --- Input reading ---
    // The input is a string representation of a 2D integer array.
    // Example: [[1,2,3],[3,2,1]]
    // Parsing this format in C++ requires manual string manipulation.
    std::string input_line;
    std::getline(std::cin, input_line);

    // --- Solution ---
    // The following variables are available:
    // input_line: std::string (e.g., "[[1,2,3],[3,2,1]]")
    // You will need to parse this string into a 2D vector of integers.

    // TODO: Implement the solution, including parsing input_line
}

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
    solve();
    return 0;
}

File: stage1/problems/problem20/starter.py
import sys
import json

def solve():
  """
  Reads input from stdin, solves the problem, and prints the output.
  """
  try:
    # --- Input reading ---
    # The input is a string representation of a 2D integer array.
    # Example: [[1,2,3],[3,2,1]]
    accounts = json.loads(sys.stdin.read())
    
    # --- Solution ---
    # The following variables are available:
    # accounts: list of lists of ints (e.g., [[1,2,3],[3,2,1]])
    
    # TODO: Implement the solution
    
  except (ValueError, json.JSONDecodeError):
    # Handle potential errors during JSON parsing or if input is not valid
    pass

if __name__ == "__main__":
    solve()